lib/instrumentation/winston.js

```diff
@@ -33,145 +34,72 @@ module.exports = function instrument(agent, winston, _, shim) {
         return createLogger.apply(this, args)
       }
 
-      registerFormatter({ opts, config, agent, winston })
+      createModuleUsageMetric('winston', agent.metrics)
+
+      if (isLocalDecoratingEnabled(config) || isMetricsEnabled(config)) {
+        registerFormatter({ opts, agent, winston })
+      }
+
+      const winstonLogger = createLogger.apply(this, args)
 
-      return createLogger.apply(this, args)
+      if (isLogForwardingEnabled(config, agent)) {
+        winstonLogger.add(new NrTransport({ agent }))
+      }
+
+      return winstonLogger
     }
   })
 }
 
 /**
- * There is no right way to do this.  But since we are automagical
```

Comment:
> There is no right way to do this.

This new way definitely feels more right!

Reply:
yes it is in fact the _right_ way based on how winston is architected.  It took me a few iterations to get that though.